src/azure-cli/azure/cli/command_modules/backup/custom.py

```diff
@@ -541,11 +548,42 @@ def list_recovery_points(cmd, client, resource_group_name, vault_name, item, sta
     if use_secondary_region:
         client = recovery_points_crr_cf(cmd.cli_ctx)
 
-    # Get recovery points
-    recovery_points = client.list(vault_name, resource_group_name, fabric_name, container_uri, item_uri, filter_string)
-    paged_recovery_points = _get_list_from_paged_response(recovery_points)
+    if recommended_for_archive:
+        client = recovery_points_recommended_cf(cmd.cli_ctx)
+        recovery_points = client.list(vault_name, resource_group_name, fabric_name, container_uri, item_uri)
+        paged_recovery_points = _get_list_from_paged_response(recovery_points)
+
+    else:
+        recovery_points = client.list(vault_name, resource_group_name, fabric_name, container_uri, item_uri,
+                                      filter_string)
+        paged_recovery_points = _get_list_from_paged_response(recovery_points)
+
+    common.fetch_tier(paged_recovery_points)
+
+    recovery_point_list = common.check_rp_move_readiness(paged_recovery_points, target_tier, is_ready_for_move)
```

Comment:
is it possible to check move readiness for a recommended RP ? did you test something like this ? 

I'm not sure whether we should allow move-readiness, tier based filtering on recommended RPs

Reply:
Currently, we are fetching the recommended RPs, and then proceeding in the normal way i.e filtering based on move-readiness, tier, etc depending on what is passed.

Reply:
I mean recommendedRPs is a new API - is REST api even returning moveReadinessInfo in the first place ? Check once with Debpriya Bose on this (I think adding these filters for recommendation command was not in the requirement).

Reply:
So, me and Akshay saw the output of recommended-for-archive (and I have also attached the ss of it here), and in it, we are getting non-null entries of other archive parameters that i created. So, I think it's safe to allow  move-readiness, tier based filtering on recommended RPs.
Also, we have raised an InvalidArgumentValueError, if recommended_for_archive is provided and is_ready_for_move is False  (after conferring with Kartik), as it would anyways return an empty set.

![img](https://user-images.githubusercontent.com/45140396/123423314-07c87b00-d5dd-11eb-90a8-e9c52f94760e.PNG)

Reply:
Got it. While I'm inline with your thinking, I would like you to check once with Debpriya Bose on this